cmd/kubectl-gadget/utils/trace.go

```diff
@@ -438,6 +438,20 @@ RetryLoop:
 	}
 }
 
+// sigpipeHandler installs a handler for SIGPIPE.
+// On reception of this signal, the given trace will be deleted.
+// This function fixes trace not being deleted when calling:
+// kubectl gadget process-collector -A | head -n0
+func sigpipeHandler(traceID string) {
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt, syscall.SIGPIPE)
+	go func() {
+		<-c
+		DeleteTrace(traceID)
+		os.Exit(1)
+	}()
+}
+
```

Comment:
I think it's a very nasty issue, I have been thinking and came up with the following points to discuss:

1. It's not only about SIGPIPE, I tried and we also have the same problem with SIGINT and probably with all other signals that cause the program to terminate. 
2. We should add the signal handler before we create the trace. (Using it only in `PrintTraceOutputFromStream` and `PrintTraceOutputFromStatus` leaves a window where we could leak a trace)
3. `DeleteTrace(traceID)` could hang up (image something is going on with the network / api server). It could be very annoying for a user pressing Ctrl + C without having any effect.

I think the solution should have a timeout to avoid waiting too much on `DeleteTrace()`. Also, for SIGINT it'll be nice if it provides a way to exit after the user presses Ctrl + C for a second time. 

Pinging other folks to see their opinions @alban, @blanquicet, @marga-kinvolk

Reply:
I modified the current implementation to deal particularly with SIGINT.
I am not really into adding a timeout for `DeleteTrace()` because we already have function with time out and I would like to avoid adding another one.